engines/mus.cpp

```diff
@@ -146,25 +150,48 @@ namespace pono {
 
     /*
      * If the conjunct encodes a state update constraint (i.e. of form
-     * <stateVar>.next = ...), then uses that state variable's id for the
+     * <stateVar>.next = ...), then use that state variable's id for the
      * control var symbol. Otherwise use the conjunct's to_string as an
      * identifier.
      */
-    UnorderedTermMap transIdToConjunct;
+    unordered_map<string, Term> transIdToConjunct;
     for (auto &tc: transConjuncts) {
       Term id = tc;
       if (tc->get_op() == PrimOp::Equal) {
         Term lhs = *tc->begin();
         if (ts_.is_next_var(lhs)) {
-          id = lhs;
+          id = ts_.curr(lhs);
         }
       }
       Term t = unrollUntilBound(tc, k);
       if (!options_.mus_include_yosys_internal_netnames_ && isYosysInternalNetname(id)) {
         solver_->assert_formula(t);
         assertions.push_back(t);
       } else {
-        transIdToConjunct.insert({id, t});
+        transIdToConjunct.insert({id->to_string(), t});
+      }
+    }
+
+    if (!options_.mus_combine_suffix_.empty()) {
+      /*
+       * Conjoin TRANS constraints that are identical up to suffix matching the
+       * supplied regular expression.
+       */
+      std::multimap<string, Term> mm;
+      for (auto & [id, tc] : unordered_map(transIdToConjunct)) {
```

Comment:
Is the call to the `unordered_map` constructor here necessary? It might be, I'm not super familiar with C++.

Reply:
Yea, the constructor call makes a copy of `transIdToConjuct` which is needed to allow removing elements from the original map while iterating over it (line 186).

Reply:
Ahh, got it. I was aware that it was a copy constructor, but I wasn't seeing the reason. Thanks for clarifying.

Good catch! I'm so out of practice with (non-Rust) manual memory management that I probably would've messed this up :)